http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -503,6 +503,19 @@ class RuntimeServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
       _ <- if (updatedAutopauseThreshold != runtime.autopauseThreshold)
         clusterQuery.updateAutopauseThreshold(runtime.id, updatedAutopauseThreshold, ctx.now).transaction.void
       else Async[F].unit
+
+      // Deleting labels
+      existingLabels <- labelQuery.getAllForResource(runtime.id, LabelResourceType.runtime).transaction
+
+      _ <- req.labelsToDelete.traverse { label =>
```

Comment:
It might be better to do all the label operations (upserts and deletes) in 1 transaction if possible.

For example we could define deletion by batch in `LabelComponent`:
```
  def deleteForResource(resourceId: Long, resourceType: LabelResourceType, keys: Set[String])(
    ): DBIO[Int] =
    labelQuery
      .filter(_.resourceId === resourceId)
      .filter(_.resourceType === resourceType)
      .filter(_.key inSetBind keys)
      .delete
```

(I think upsert needs to be done one by one.)

Then here we could do:
```
      dbOp = DBIO.seq(
        labelQuery.deleteForResource(runtime.id, LabelResourceType.runtime, req.labelsToDelete),
        req.labelsToUpsert.toList.traverse_ {
          case (k, v) =>
            labelQuery.save(runtime.id, LabelResourceType.runtime, k, v)
        }
      )
      _ <- dbOp.transaction
```